### PR TITLE
CLI: Updates `tkn hub install task` cli command for --version flag to add deprecation warning

### PR DESCRIPTION
**Pull request by pratap0007**
# Changes

This patch adds deprecation warning when users install
a deprecation version of a resource

Signed-off-by: Shiv Verma <shverma@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [x] Run API Unit Tests, Lint Checks, API Design, Golden Files with `make api-check`
- [ ] Run UI Unit Tests, Lint Checks with `make ui-check`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

### CODE REVIEW

**Comment by piyush-garg** [comment]
/hold

Needs to be merged after #284

File: api/pkg/cli/cmd/install/install.go [hunk 1]
```diff
@@ -153,6 +154,15 @@ func (opts *options) run() error {
 		}
 	}
 
+	var deprecated bool
+	if opts.resource.GetAnnotations()[deprecationAnnotation] == "true" {
+		deprecated = true
+	}
+
+	if deprecated {
+		_ = printer.New(out).String("WARN: this version has been deprecated")
```
Comment (PuneetPunamiya) [inline api/pkg/cli/cmd/install/install.go:163]: `_ = printer.New(out).String("WARN: This version has been deprecated")`

Can you please do this nit change

File: api/pkg/cli/cmd/install/install.go [hunk 2]
```diff
@@ -153,6 +154,15 @@ func (opts *options) run() error {
 		}
 	}
 
+	var deprecated bool
+	if opts.resource.GetAnnotations()[deprecationAnnotation] == "true" {
+		deprecated = true
```
Comment (PuneetPunamiya) [inline api/pkg/cli/cmd/install/install.go:159]: Are we using the variable anywhere again, if not can't we directly log here ??

File: api/pkg/cli/cmd/install/install_test.go [hunk 3]
```diff
@@ -59,6 +59,36 @@ var resVersion = &res.ResourceVersionData{
 	},
 }
 
+var deprecated = true
+var resDeprecatedVersion = &res.ResourceVersionData{
+	ID:                  11,
+	Version:             "0.2",
+	DisplayName:         "foo-bar",
+	Description:         "v0.2 Task to run foo",
+	Deprecated:          &deprecated,
```
Comment (PuneetPunamiya) [inline api/pkg/cli/cmd/install/install_test.go:68]: Can't we directly use  `true` here ??

File: api/pkg/cli/cmd/install/install.go (hunk 1 above)
Comment (PuneetPunamiya) [reply api/pkg/cli/cmd/install/install.go:163]: Should we also use a `bulb` 💡 or something while logging the warning ?
cc @sm43  @piyush-garg

File: api/pkg/cli/cmd/install/install_test.go (hunk 3 above)
Comment (pratap0007) [reply api/pkg/cli/cmd/install/install_test.go:68]: No, we can't use directly

File: api/pkg/cli/cmd/install/install.go (hunk 2 above)
Comment (pratap0007) [reply api/pkg/cli/cmd/install/install.go:159]: no,it needs a boolean

File: api/pkg/cli/cmd/install/install_test.go (hunk 3 above)
Comment (PuneetPunamiya) [reply api/pkg/cli/cmd/install/install_test.go:68]: Why ??

File: api/pkg/cli/cmd/install/install.go (hunk 2 above)
Comment (PuneetPunamiya) [reply api/pkg/cli/cmd/install/install.go:159]: I think this should work 

```
	if opts.resource.GetAnnotations()[deprecationAnnotation] == "true" {
		_ = printer.New(out).String("WARN: this version has been deprecated")
	}
```

File: api/pkg/cli/cmd/install/install_test.go (hunk 3 above)
Comment (pratap0007) [reply api/pkg/cli/cmd/install/install_test.go:68]: it's an optional field so it takes an address

Comment (PuneetPunamiya) [reply api/pkg/cli/cmd/install/install_test.go:68]: Ahh right I got it, it's the optional value and hence needs to be a pointer 👍🏻

**Pull request merged**